lib/puppet/ssl/host.rb

```diff
@@ -395,43 +395,37 @@ def state
 
   private
 
-  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  # Ensures that CRL is on disk; downloads the CRL from server if not found
+  # Does nothing if use_crl? is set not truthy
   def ensure_crl_if_needed
     if use_crl? && !Puppet::FileSystem.exist?(crl_path)
-      # The CertificateRevocationList indirector will attempt to download the
-      # CRL from the CA if it does not exist on disk. It will ask host for
-      # its ssl_store again, but expect crl checking to be disabled for that
-      # store. This is not thread safe, and should be replaced as soon as we
-      # no longer need to use the indirector to download the CRL (PUP-8654).
-      old_crl_usage = @crl_usage
-      @crl_usage = false
-      Puppet.debug _("Disabling certificate revocation checking when fetching the CRL and no CRL is present")
-      if !CertificateRevocationList.indirection.find(CA_NAME)
-        raise Puppet::Error,
-              _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
-      end
-      @crl_usage = old_crl_usage
+      download_and_save_crl_bundle
     end
   end
 
-  # @param path [String] Path to CRL Chain
+  # @param crl_string [String] CRLs read from disk or obtained from server
   # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
-  # @raise [Errno::ENOENT] if file does not exist
   # @raise [Puppet::Error<OpenSSL::X509::CRLError>] if the CRL chain is malformed
-  def load_crls(path)
+  def process_crl_string(crl_string)
     delimiters = /-----BEGIN X509 CRL-----.*?-----END X509 CRL-----/m
-    crls_pems = Puppet::FileSystem.read(path, encoding: Encoding::UTF_8)
-    crls_pems.scan(delimiters).map do |crl|
+    crl_string.scan(delimiters).map do |crl|
       begin
         OpenSSL::X509::CRL.new(crl)
       rescue OpenSSL::X509::CRLError => e
         raise Puppet::Error.new(
-            _("Failed attempting to load CRL from %{crl_path}! The CRL below caused the error '%{error}':\n%{crl}" % {crl_path: crl_path, error: e.message, crl: crl}),
+          _("Failed attempting to load CRL from %{crl_path}! The CRL below caused the error '%{error}':\n%{crl}" % {crl_path: crl_path, error: e.message, crl: crl}),
           e)
       end
     end
   end
 
+  # @param path [String] Path to CRL Chain
+  # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
```

Comment:
This will raise `Errno::ENOENT` if the file doesn't exist, I think, per the doc that was on line 419 above.

Reply:
Does this need that documentation here to? I assumed that sort of documentation is necessary if this method is explicitly raising those exceptions.

Reply:
yeah